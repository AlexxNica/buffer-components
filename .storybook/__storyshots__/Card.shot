exports[`Default`] = `
<div
  className="">
  What is a Product Designer? An awesome story by 
  <a
    href="#">
    @jgadapee
  </a>
   over on Medium! 
  <a
    href="#">
    http://buff.ly/1LTbUqv
  </a>
</div>
`;

exports[`doublePadding`] = `
<div
  className=" ">
  What is a Product Designer? An awesome story by 
  <a
    href="#">
    @jgadapee
  </a>
   over on Medium! 
  <a
    href="#">
    http://buff.ly/1LTbUqv
  </a>
</div>
`;

exports[`empty`] = `
<div
  className=" ">
  Nothing to see here! ðŸ‘»
</div>
`;

exports[`failed`] = `
<div
  className=" ">
  What is a Product Designer? An awesome story by 
  <a
    href="#">
    @jgadapee
  </a>
   over on Medium! 
  <a
    href="#">
    http://buff.ly/1LTbUqv
  </a>
</div>
`;

exports[`noBorder`] = `
<div
  className=" ">
  What is a Product Designer? An awesome story by 
  <a
    href="#">
    @jgadapee
  </a>
   over on Medium! 
  <a
    href="#">
    http://buff.ly/1LTbUqv
  </a>
</div>
`;

exports[`noPadding`] = `
<div
  className=" ">
  What is a Product Designer? An awesome story by 
  <a
    href="#">
    @jgadapee
  </a>
   over on Medium! 
  <a
    href="#">
    http://buff.ly/1LTbUqv
  </a>
</div>
`;
