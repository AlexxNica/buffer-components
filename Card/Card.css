@import '../layout.css';
@import '../variables.css';

/* Card */
.Card {
  margin-bottom: 16px;
  padding: 16px;
  background-color: var(--white);
  border: 1px solid var(--geyser);
  border-radius: var(--border-radius-level-1);
}

.Card .Card {
  border-radius: var(--border-radius-level-2);
}

.Card .Card .Card {
  border-radius: var(--border-radius-level-3);
}

/* Card - Modifiers */
.Card--double-padding {
  padding: 32px;
}

.Card--empty {
  border-style: dashed;
}

.Card--no-border {
  border: 0;
}

.Card--no-padding {
  padding: 0;
}
