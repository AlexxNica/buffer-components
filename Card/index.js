import React, { PropTypes } from 'react';
import { classNames, cleanProps } from '../lib/utils';
import styles from './card.css';

const Card = (props) => {
  const classes = classNames(styles, 'card', {
    'double-padding': props.doublePadding,
    empty: props.empty,
    failed: props.failed,
    'no-border': props.noBorder,
    'no-padding': props.noPadding,
  });
  return (
    <div className={classes} {...cleanProps(props, Card.propTypes)}>
      {props.children}
    </div>
  );
};

Card.propTypes = {
  children: PropTypes.node,
  doublePadding: PropTypes.bool,
  empty: PropTypes.bool,
  failed: PropTypes.bool,
  noBorder: PropTypes.bool,
  noPadding: PropTypes.bool,
};

export default Card;
