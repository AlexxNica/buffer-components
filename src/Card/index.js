import React, { PropTypes } from 'react';
import reactCSS from 'reactcss';

const Card = ({ children, empty, noBorder }) => {
  const style = reactCSS({
    default: {
      Card: {
        backgroundColor: '#fff',
        border: '1px solid #ced7df',
        borderRadius: 2,
        padding: 16,
      },
    },
    empty: {
      Card: {
        borderStyle: 'dashed',
      },
    },
    noBorder: {
      Card: {
        border: 0,
      },
    },
  }, {
    empty,
    noBorder,
  });
  return (
    <div
      style={style.Card}
    >
      {children}
    </div>
  );
};

Card.propTypes = {
  children: PropTypes.node,
  empty: PropTypes.bool,
  noBorder: PropTypes.bool,
};

export default Card;
