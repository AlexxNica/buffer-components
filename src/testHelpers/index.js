export const renderAndCheckSnapshot = (component) => {
  const renderedComponent = renderer.create(component);
  const tree = renderedComponent.toJSON();
  expect(tree).toMatchSnapshot();
};
