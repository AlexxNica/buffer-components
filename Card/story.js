import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Card from './index';

storiesOf('Card')
  .add('Default', () => (
    <Card>A Regular Old Card</Card>
  ))
  .add('doublePadding', () => (
    <Card doublePadding>A Card With Extra Padding</Card>
  ))
  .add('empty', () => (
    <Card empty>An Empty Old Card</Card>
  ))
  .add('noBorder', () => (
    <Card noBorder>A Borderless Old Card</Card>
  ))
  .add('noPadding', () => (
    <Card noPadding>A Card With No Padding</Card>
  ));
