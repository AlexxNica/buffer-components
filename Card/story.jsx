import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Card from './index';

storiesOf('Card')
  .add('Default', () => (
    <Card>What is a Product Designer? An awesome story by <a href="#">@jgadapee</a> over on Medium! <a href="#">http://buff.ly/1LTbUqv</a></Card>
  ))
  .add('doublePadding', () => (
    <Card doublePadding>What is a Product Designer? An awesome story by <a href="#">@jgadapee</a> over on Medium! <a href="#">http://buff.ly/1LTbUqv</a></Card>
  ))
  .add('empty', () => (
    <Card empty>Nothing to see here! ðŸ‘»</Card>
  ))
  .add('failed', () => (
    <Card failed>What is a Product Designer? An awesome story by <a href="#">@jgadapee</a> over on Medium! <a href="#">http://buff.ly/1LTbUqv</a></Card>
  ))
  .add('noBorder', () => (
    <Card noBorder>What is a Product Designer? An awesome story by <a href="#">@jgadapee</a> over on Medium! <a href="#">http://buff.ly/1LTbUqv</a></Card>
  ))
  .add('noPadding', () => (
    <Card noPadding>What is a Product Designer? An awesome story by <a href="#">@jgadapee</a> over on Medium! <a href="#">http://buff.ly/1LTbUqv</a></Card>
  ));
