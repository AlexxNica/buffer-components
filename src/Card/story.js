import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Card from './index';

storiesOf('Card')
  .add('Default', () => (
    <Card>A Regular Old Card</Card>
  ))
  .add('Empty', () => (
    <Card empty>An Empty Old Card</Card>
  ))
  .add('No Border', () => (
    <Card noBorder>A Borderless Old Card</Card>
  ));
