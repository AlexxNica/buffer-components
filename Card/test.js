jest.unmock('./index');
import React from 'react';
import { renderAndCheckSnapshot } from '../testHelpers';
import Card from './index';

describe('Card', () => {
  it('Card component', () => {
    renderAndCheckSnapshot(<Card>Some Contents</Card>);
  });

  it('Empty Card', () => {
    renderAndCheckSnapshot(<Card empty>Some Contents</Card>);
  });

  it('Borderless Card', () => {
    renderAndCheckSnapshot(<Card noBorder>Some Contents</Card>);
  });
});
