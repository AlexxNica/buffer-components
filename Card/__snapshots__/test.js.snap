exports[`Card Borderless Card 1`] = `
<div
  style={
    Object {
      "MozBorderRadius": 2,
      "OBorderRadius": 2,
      "WebkitBorderRadius": 2,
      "backgroundColor": "#fff",
      "border": 0,
      "borderRadius": 2,
      "msBorderRadius": 2,
      "padding": 16
    }
  }>
  Some Contents
</div>
`;

exports[`Card Card component 1`] = `
<div
  style={
    Object {
      "MozBorderRadius": 2,
      "OBorderRadius": 2,
      "WebkitBorderRadius": 2,
      "backgroundColor": "#fff",
      "border": "1px solid #ced7df",
      "borderRadius": 2,
      "msBorderRadius": 2,
      "padding": 16
    }
  }>
  Some Contents
</div>
`;

exports[`Card Empty Card 1`] = `
<div
  style={
    Object {
      "MozBorderRadius": 2,
      "OBorderRadius": 2,
      "WebkitBorderRadius": 2,
      "backgroundColor": "#fff",
      "border": "1px solid #ced7df",
      "borderRadius": 2,
      "borderStyle": "dashed",
      "msBorderRadius": 2,
      "padding": 16
    }
  }>
  Some Contents
</div>
`;
